tasks/kernel_matrix_testing/README.md

```diff
@@ -1,425 +1,546 @@
+# Check status of all VMs
+dda inv -e kmt.status
+
+# When finished, destroy all resources
+dda inv -e kmt.destroy-stack
 ```
 
-This will attempt to manually teardown all resources. Primarily we care about cleaning the local libvirt environment and destroy remote ec2 instances.
+## Core Concepts
+
+### Stacks
+
+The system works on the concept of `stacks`. A `stack` is a collection of VMs, both local and remote, that are managed together. A `stack` is given a unique name by the user. If no stack name is provided, a name is generated automatically from the current git branch name. This allows developers to couple `stacks` with their git workflow.
+
+A `stack` can be:
 
-## Tasks
+-   Created
+-   Configured
+-   Launched
+-   Paused
+-   Resumed
+-   Destroyed
 
-### Initializing environment
+### VMsets File
 
-If you only want to initialize the directory structure, you may do a light setup without any parameters:
+Configuring a stack involves generating a configuration file, referred to as the `vmsets` file. This JSON file specifies the sets of VMs to launch. A "set" of VMs is a collection of VMs sharing characteristics like architecture or purpose (e.g., "all x86_64 distribution VMs on a remote machine").
+
+This file is managed by the `kmt.gen-config` command. You typically don't need to edit it manually.
+
+A sample VMSet file can be found [here](https://github.com/DataDog/test-infra-definitions/blob/f85e7eb2f003b6f9693c851549fbb7f3969b8ade/scenarios/aws/microVMs/sample-vm-config.json).
+
+### Local vs. Remote VMs
+
+KMT can manage two types of VMs:
+
+-   **Local VMs**: These run directly on your machine using `libvirt` and `qemu`. They will have the same architecture as your host machine.
+-   **Remote VMs**: These are EC2 instances launched in AWS. KMT can manage both x86_64 and arm64 remote VMs. Launching remote VMs requires configuring an SSH key.
+
+## Command Reference
+
+This section provides a detailed reference for all `kmt` tasks.
+
+### `kmt.init`
+
+Initializes the KMT environment. It downloads VM images and tools, and installs system dependencies. This command only needs to be run once.
 
 ```bash
-dda inv -e kmt.init
+# Initialize with specific VM images
+dda inv -e kmt.init --images=ubuntu_22.04,debian_10
+
+# Initialize and download all available VM images for your architecture
+dda inv -e kmt.init --all-images
+
+# If you only intend to manage remote VMs
+dda inv -e kmt.init --remote-setup-only
 ```
 
-### Updating resources
+The `--images` parameter is required unless `--remote-setup-only` or `--all-images` is specified, or you explicitly confirm the interactive prompt. Use `kmt.ls` to see available images.
 
-In order to update the resources for launching VMs run:
+This command will also guide you through the [default SSH key configuration](#kmtconfig-ssh-key).
+
+### `kmt.update-resources`
+
+Updates the resources for launching VMs, such as VM images and tools.
 
 ```bash
 dda inv -e kmt.update-resources
```

Comment:
We should add an example usage of `--images` here as well to show that specific VM images can be downloaded with this task.